examples/https_client.cpp

```diff
@@ -37,25 +37,28 @@ class https_client {
     // TODO: Consider async connect
     net::connect(socket_.next_layer(), endpoint_iterator);
 
-    // TODO: Implement async handshake
-    socket_.handshake(ssl::stream_base::client);
-
-    std::ostream request_stream(&request_);
-    request_stream << "GET " << path << " HTTP/1.0\r\n";
-    request_stream << "Host: " << host << "\r\n";
-    request_stream << "Accept: */*\r\n";
-    request_stream << "Connection: close\r\n\r\n";
-
-    boost::asio::async_write(socket_, request_, [this](const boost::system::error_code& ec, std::size_t) {
+    socket_.async_handshake(ssl::stream_base::client, [this, &path, &host](const boost::system::error_code& ec) {
       if (ec) {
-        std::cerr << "Error sending: request:" << ec.message() << "\n";
+        std::cerr << "Error handshaking:" << ec.message() << "\n";
         return;
       }
-      net::async_read_until(socket_, response_, "\r\n", [this](boost::system::error_code ec, size_t) {
+      std::ostream request_stream(&request_);
+      request_stream << "GET " << path << " HTTP/1.0\r\n";
+      request_stream << "Host: " << host << "\r\n";
+      request_stream << "Accept: */*\r\n";
+      request_stream << "Connection: close\r\n\r\n";
```

Comment:
HTTP/1.0 implies `Connection: close`

Reply:
Indeed it does. This is in no way anywhere close to being a complete HTTP client, but I might as well remove this line.

I would like to rewrite this example to use `boost::beast` I think. And potentially use it to make some integration tests with some real servers.